### ISSUE TITLE
Ensure devhub worker tasks are cleaning up local temporary files

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#16879]
We had an issue where `/` filed up in production on devhub celery workers. 

`repack_fileupload` is the most likely suspect as it writes to the local tempdir (normally `/tmp`) which isn't on a separate filesystem. It _should_ cleanup after itself, but maybe there is an issue with that. Or maybe it's a completely different task that's the problem and we just happened to notice it because uploads were broken and we were left with errors about repack.

We should verify which task is actually responsible and fix it.

**Comment by diox** [comment]
The error we had saying no space left on device : https://sentry.prod.mozaws.net/operations/olympia-prod/issues/11006595/?query=is%3Aunresolved

Note that there should have been an ops alert too, I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1703183 for this.

**Comment by diox** [comment]
`repack_fileupload` specifically uses `move_stored_file`:
https://github.com/mozilla/addons-server/blob/593e0331e7ff4fd290f5bc7891da1cacd5877188/src/olympia/files/tasks.py#L45-L51

`move_stored_file` should delete after itself:
https://github.com/mozilla/addons-server/blob/593e0331e7ff4fd290f5bc7891da1cacd5877188/src/olympia/amo/storage_utils.py#L80

Doesn't mean it's working though, we should check that. If that works, maybe it's another part of the validation process that's problematic, or maybe another task entirely that happens to share the same workers (image processing is possible). Knowing what kind of files `/tmp` was filled with would help.

**Comment by willdurand** [comment]
I don't see `tempdir` being cleaned up here: https://github.com/mozilla/addons-server/blob/c121e3533f44d35f78c811e4552094f423fff8ee/src/olympia/files/tasks.py#L32

**Comment by diox** [comment]
Yes, but that shouldn't be too much of an issue ? The file itself should be moved.

**Comment by willdurand** [comment]
That depends on the FS but maybe we run out of inodes.

**Comment by diox** [comment]
So, from looking at stage, we know that `/tmp` has a bunch of temporary directories with some add-on files extracted in them (not `zip`/`xpi` files themselves, but their contents). Looking at the `repack_fileupload` again, as mentioned above we call `move_stored_file()` on the zip/xpi, but we don't see to ever clean up the contents after they have been extracted and repacked, and I think that's the source of our problems.

**Comment by diox** [comment]
We'll check this one ourselves once it lands.
